### PR TITLE
chore: always fully generate the repo if it is a non-monorepo or contains common protos

### PR DESCRIPTION
**Pull request by JoeWang1127**
In this PR:
- Always perform a full generation in a non-monorepo or the repo contains common protos.
- Secure generation workflow (use environment variable to avoid script injection), inspired by https://github.com/googleapis/java-bigtable/pull/2317

This PR also brings changes in common protos and iam due to protoc updates (25.3 -> 25.4).

### CODE REVIEW

**Comment by blakeli0** [comment]
> This PR also brings changes in common protos and iam due to protoc updates (25.3 -> 25.4).

Can we leave this part to the nightly generation?

**Comment by JoeWang1127** [comment]
> Can we leave this part to the nightly generation?

When I changed the workflow, the workflow brings this change. It's not a manual commit.

I think it's good because it verified the expected behavior.

File: library_generation/test/cli/entry_point_unit_tests.py [hunk 1]
```diff
@@ -138,3 +138,62 @@ def test_generate_non_monorepo_with_changes_triggers_full_generation(
...
+        current_config_path = (
+            f"{test_resource_dir}/monorepo_without_common_protos_modified.yaml"
+        )
+        self.assertTrue(from_yaml(current_config_path).is_monorepo())
+        self.assertTrue(from_yaml(baseline_config_path).is_monorepo())
+        # we call the implementation method directly since click
+        # does special handling when a method is annotated with @main.command()
+        generate_impl(
+            baseline_generation_config_path=baseline_config_path,
+            current_generation_config_path=current_config_path,
```
Comment (blakeli0) [inline library_generation/test/cli/entry_point_unit_tests.py:194]: Can we use the same config path? I think it should not matter if the two configs are the same?

File: library_generation/cli/entry_point.py [hunk 2]
```diff
@@ -163,6 +163,15 @@ def __generate_repo_and_pr_description_impl(
     )
 
 
+def __needs_full_repo_generation(config_change: ConfigChange) -> bool:
```
Comment (diegomarquezp) [inline library_generation/cli/entry_point.py:166]: nit: private functions are [recommended](https://g3doc.corp.google.com/eng/doc/devguide/py/style/index.md?cl=head#316-naming) to have single underscores

**Review by diegomarquezp — COMMENTED** [review]
LGTM, just a nit.

File: library_generation/test/cli/entry_point_unit_tests.py (hunk 1 above)
Comment (JoeWang1127) [reply library_generation/test/cli/entry_point_unit_tests.py:194]: This test is verifying the list of changed library can be populated in a monorepo.

If we use two identical configs, the list should be empty, i.e., no changed library. I think this is fine as long as the list is not None.

I'll make the change.

File: library_generation/cli/entry_point.py (hunk 2 above)
Comment (JoeWang1127) [reply library_generation/cli/entry_point.py]: I changed the function name in this file.

We can gradually change functions in other files in future PRs.

File: library_generation/test/generate_library_unit_tests.py [hunk 3]
```diff
@@ -66,13 +79,13 @@ def _run_command_and_get_sdout(self, command, **kwargs):
     def test_get_grpc_version_with_no_env_var_fails(self):
         # the absence of DOCKER_GRPC_VERSION will make this function to fail
         result = self._run_command("get_grpc_version")
-        self.assertEquals(1, result.returncode)
```
Comment (JoeWang1127) [inline library_generation/test/generate_library_unit_tests.py:69]: `assertEquals` is deprecated so I changed it to `assertEqual`.

**Review by diegomarquezp — APPROVED** [review]

**Review by blakeli0 — APPROVED** [review]

File: library_generation/cli/entry_point.py (hunk 2 above)
Comment (blakeli0) [reply library_generation/cli/entry_point.py]: Thanks @diegomarquezp, TIL.

**Pull request merged**